packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/util.ts

```diff
@@ -99,3 +102,24 @@ export function getDefaultFigureForType(type: LockedFigureType): LockedFigure {
             throw new UnreachableCaseError(type);
     }
 }
+
+export function generateLockedFigureAppearanceDescription(
+    color: LockedFigureColor,
+    strokeStyle: LockedLineStyle = "solid",
+    fill?: LockedFigureFillType,
+) {
+    const convertedColor = color === "grayH" ? "gray" : color;
+    const convertedFill = fill === "none" ? "no" : `${fill}`;
+
+    switch (fill) {
+        case "none":
+            return `. Appearance ${strokeStyle} ${convertedColor} border, with ${convertedFill} fill.`;
+        case "white":
+            return `. Appearance ${strokeStyle} ${convertedColor} border, with a ${convertedFill} fill.`;
+        case "solid":
+        case "translucent":
+            return `. Appearance ${strokeStyle} ${convertedColor} border, with a ${convertedFill} ${convertedColor} fill.`;
+        default:
+            return `. Appearance ${strokeStyle} ${convertedColor}.`;
```

Comment:
Since we're switching on `fill` here, is `convertedFill` needed? I.e. could this become

```suggestion
    switch (fill) {
        case "none":
            return `. Appearance ${strokeStyle} ${convertedColor} border, with no fill.`;
        case "white":
            return `. Appearance ${strokeStyle} ${convertedColor} border, with a white fill.`;
        case "solid":
        case "translucent":
            return `. Appearance ${strokeStyle} ${convertedColor} border, with a ${fill} ${convertedColor} fill.`;
        default:
            return `. Appearance ${strokeStyle} ${convertedColor}.`;
```

Also, if the `default` case is meant to handle `undefined`, it might be good to handle `undefined` explicitly and add an [exhaustiveness check](https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript) to the default branch:

```ts
    case undefined:
        return `. Appearance ${strokeStyle} ${convertedColor}.`;
    default:
        throw new UnreachableCaseError(fill);
```

This will ensure that we get a type error here, reminding us to update this code, if new fill types are ever added.